grpc/execution/server.go

```diff
@@ -252,18 +265,54 @@ func (s *ExecutionServiceServerV1Alpha2) ExecuteBlock(ctx context.Context, req *
 				log.Debug("ignoring deposit tx with invalid asset ID", "assetID", deposit.AssetId)
 				continue
 			}
+
 			assetID := [32]byte{}
 			copy(assetID[:], deposit.AssetId[:32])
+
 			if _, ok := s.bridgeAllowedAssetIDs[assetID]; !ok {
 				log.Debug("ignoring deposit tx with disallowed asset ID", "assetID", deposit.AssetId)
 				continue
 			}
 
-			amount := protoU128ToBigInt(deposit.Amount)
-			address := common.HexToAddress(deposit.DestinationChainAddress)
+			recipient := common.HexToAddress(deposit.DestinationChainAddress)
+			amount := bac.ScaledDepositAmount(protoU128ToBigInt(deposit.Amount))
+
+			if bac.Erc20Asset != nil {
+				log.Debug("creating deposit tx to mint ERC20 asset", "token", bac.AssetDenom, "erc20Address", bac.Erc20Asset.ContractAddress)
+				abi, err := contracts.AstriaMintableERC20MetaData.GetAbi()
+				if err != nil {
+					// this should never happen, as the abi is hardcoded in the contract bindings
+					return nil, fmt.Errorf("failed to get abi for erc20 contract for asset %s: %w", bac.AssetDenom, err)
+				}
+
+				// pack arguments for calling the `mint` function on the ERC20 contract
+				args := []interface{}{recipient, amount}
+				calldata, err := abi.Pack("mint", args...)
+				if err != nil {
+					return nil, err
+				}
+
+				txdata := types.DepositTx{
+					From:  s.bridgeSenderAddress,
+					Value: new(big.Int), // don't need to set this, as we aren't minting the native asset
+					// mints cost ~14k gas, however this can vary based on existing storage, so we add a little extra as buffer.
+					//
+					// the fees are spent from the "bridge account" which is not actually a real account, but is instead some
+					// address defined by consensus, so the gas cost is not actually deducted from any account.
+					Gas:  16000,
```

Comment:
I'm not sure if this is a big enough number, the gas costs of a tx is normally gasUnits * gasPrice. I'd either make it so the DepositTxs pay no gas or set this to something crazy large to account for EIP1559 gas price swings

Reply:
good point, I'll make it higher for now since this sender "account" has infinite funds. I originally tried to just make the tx not use gas, but it got a bit janky when it came to the state transition gas accounting

Reply:
Lol I got that wrong, you're right it's the gas unit and not gas cost. This should be fine then